src/qbe_ir_generator.cpp

```diff
@@ -70,6 +70,12 @@ std::string GetBinaryOperator(BinaryOperator op) {
       return "xor";
     case BinaryOperator::kOr:
       return "or";
+    case BinaryOperator::kShl:
+      return "shl";
+    // Arithmetic shift right (sar) is only equivalent to a division by a power
+    // of two for non-negative numbers.
```

Comment:
Suggest making this a `NOTE:` since this is an important detail. State that _right-shifting negative numbers is implementation-defined; we choose to also use arithmetic shift on them._